### PR TITLE
[4.1] Articles filtered by me

### PR DESCRIPTION
**Pull request by brianteeman**
### Summary of Changes
This PR adds the ability to filter articles by the logged in user to make it easier to find your own articles. If you are creating your own admin menu then you can also create a menu item for "My articles" by filtering the article as shown below.

### Testing Instructions
On a site with multiple articles created by multiple authors

1. Create an admin menu with an item of articles list and set the filter to "Created by me"

image

2. Use the article filters to just show your own articles

image

### CODE REVIEW

**Comment by crystalenka** [comment]
I have tested this item :white_check_mark: successfully on b029489bcdb3e31095dec84784655635a88d81dd<br /><br />You're quick! It works well. Thanks!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36555">issues.joomla.org/tracker/joomla-cms/36555</a>.</sub>

**Comment by crystalenka** [comment]
I have tested this item :white_check_mark: successfully on bb483a230bc3dcc3cb112685162001b02f4ede27<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36555">issues.joomla.org/tracker/joomla-cms/36555</a>.</sub>

File: administrator/components/com_content/src/Model/ArticlesModel.php [hunk 1]
```diff
@@ -462,6 +462,14 @@ protected function getListQuery()
 		}
 		elseif (is_array($authorId))
 		{
+			// Check to see if by_me is in the array
+			if (in_array('by_me', $authorId))
+
+			// Replace by_me with the current user id in the array
+			{
+				$authorId['by_me'] = $user->id;
```
Comment (Quy) [inline administrator/components/com_content/src/Model/ArticlesModel.php]:
```suggestion
			if (in_array('by_me', $authorId))
			{
				// Replace by_me with the current user id in the array
				$authorId['by_me'] = $user->id;
```

Comment (brianteeman) [reply administrator/components/com_content/src/Model/ArticlesModel.php]: i am happy as it is

**Comment by nikosdion** [comment]
I have tested this item :white_check_mark: successfully on 0fda4a9583342d144e7cf4e526db08be6ec28d35<br /><br />Tested on a dev site with two different user publishing articles, on different user groups even. Works flawlessly. Thanks, Brian!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36555">issues.joomla.org/tracker/joomla-cms/36555</a>.</sub>

**Comment by crystalenka** [comment]
I have tested this item :white_check_mark: successfully on 0fda4a9583342d144e7cf4e526db08be6ec28d35<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36555">issues.joomla.org/tracker/joomla-cms/36555</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36555">issues.joomla.org/tracker/joomla-cms/36555</a>.</sub>

File: administrator/components/com_content/src/Model/ArticlesModel.php [hunk 2]
```diff
@@ -462,6 +462,14 @@ protected function getListQuery()
 		}
 		elseif (is_array($authorId))
 		{
+			// Check to see if by_me is in the array
+			if (in_array('by_me', $authorId))
```
Comment (PhilETaylor) [inline administrator/components/com_content/src/Model/ArticlesModel.php]: in_array is one of those micro-optimisations 

```
"array_slice"
"assert"
"boolval"
"call_user_func"
"call_user_func_array"
"chr"
"count"
"defined"
"doubleval"
"floatval"
"func_get_args"
"func_num_args"
"get_called_class"
"get_class"
"gettype"
"in_array"
"intval"
"is_array"
"is_bool"
"is_double"
"is_float"
"is_int"
"is_integer"
"is_long"
"is_null"
"is_object"
"is_real"
"is_resource"
"is_string"
"ord"
"strlen"
"strval"
```

```suggestion
			if (\in_array('by_me', $authorId))
```

**Comment by bembelimen** [comment]
Thx

**Comment by brianteeman** [comment]
thanks

**Pull request merged**